### PR TITLE
Visual Studio 2015 compile warning/error workarounds

### PR DESCRIPTION
**Pull request by figment**
This is the list of Visual Studio 2015 changes I made to get CDDA to compile with fewer warnings and errors.  Several are warnings that might compile like double to float casts or use bad operators like + on booleans.

Major issues were the read file function casting with bind not automatically sorting out which version of the override to use.   I removed some Dynamic Array work since that seems to have changed recently so that dynamically sized stack allocated arrays are not used any more.  Though I worry about my_MAPSIZE vs MAPSIZE potential differences which could in theory lead to buffer overflows.

I changes some include locations related to SDL2 because the source libraries do not actually use that folder convention so I don't believe this project should require moving files around just so that SDL2 is prefixed.

My headers/binaries used for WinDepends folder can be found in the release notes at https://github.com/figment/Cataclysm-DDA/releases/tag/0.C-figment-v1.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
That `SDL2/` part will need checking on multiple compilers. Some of them may require that.

**Comment by Coolthulhu** [comment]
The buildbot complains about unused variables. This needs to be fixed.

Looks like that variable is declared twice in the same function, but only used once.

File: src/lightmap.cpp [hunk 1]
```diff
@@ -58,6 +58,7 @@ void map::build_transparency_cache( const int zlev )
     }
 
     // Default to just barely not transparent.
+    const float LightTransparencyOpenAir = (float)LIGHT_TRANSPARENCY_OPEN_AIR;
```
Comment (BrettDong) [inline src/lightmap.cpp]: Unused variable.

File: src/cata_utility.cpp [hunk 2]
```diff
@@ -524,7 +524,7 @@ bool read_from_file_optional( const std::string &path,
     return file_exist( path ) && read_from_file( path, reader );
 }
 
-bool read_from_file_optional( const std::string &path,
+bool read_from_file_optional_json( const std::string &path,
                               const std::function<void( JsonIn & )> &reader )
```
Comment (BrettDong) [inline src/cata_utility.cpp]: astyle regression here.

File: src/cata_utility.h [hunk 3]
```diff
@@ -151,12 +151,12 @@ class JsonDeserializer;
...
 bool read_from_file( const std::string &path, const std::function<void( std::istream & )> &reader );
-bool read_from_file( const std::string &path, const std::function<void( JsonIn & )> &reader );
+bool read_from_file_json( const std::string &path, const std::function<void( JsonIn & )> &reader );
 bool read_from_file( const std::string &path, JsonDeserializer &reader );
 
 bool read_from_file_optional( const std::string &path,
                               const std::function<void( std::istream & )> &reader );
-bool read_from_file_optional( const std::string &path,
+bool read_from_file_optional_json( const std::string &path,
                               const std::function<void( JsonIn & )> &reader );
```
Comment (BrettDong) [inline src/cata_utility.h]: astyle regression also here.

**Pull request merged**